applications/RANSApplication/custom_conditions/incompressible_potential_flow/incompressible_potential_flow_pressure_body_force_condition.cpp

```diff
@@ -0,0 +1,230 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/variables.h"
+#include "utilities/math_utils.h"
+
+// Application includes
+#include "custom_utilities/rans_calculation_utilities.h"
+#include "rans_application_variables.h"
+
+// Include base h
+#include "incompressible_potential_flow_pressure_body_force_condition.h"
+
+namespace Kratos
+{
+template <unsigned int TDim, unsigned int TNumNodes>
+int IncompressiblePotentialFlowPressureBodyForceCondition<TDim, TNumNodes>::Check(
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    int Check = BaseType::Check(rCurrentProcessInfo);
+
+    const auto& r_geometry = this->GetGeometry();
+
+    for (IndexType i_node = 0; i_node < TNumNodes; ++i_node) {
+        const auto& r_node = r_geometry[i_node];
+
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(PRESSURE_POTENTIAL, r_node);
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DENSITY, r_node);
```

Comment:
Why you need the `DENSITY` to be stored in the nodes if it is incompressible?

Reply:
Actually, there is no reason. I was following the old fluid ways. We can put them in Properties. At the moment RANS doesnt read anything from materials.json, so my plan is to change all of these to be read from it. Would it be ok if I do this change in a seperate PR? This needs to be changed in Elements as well. (As we discussed, I have to switch to DYNAMIC_VISCOSITY from KINEMATIC_VISCOSITY) :)

Reply:
OK. Let's merge this and then do all the material-related changes in one PR.